root/static/scripts/edit/URLCleanup.js

```diff
@@ -821,6 +835,15 @@ validationRules[LINK_TYPES.bbcmusic.artist] = function (url) {
   return /bbc\.co\.uk\/music\/artists\//.test(url);
 };
 
+// allow only BookBrainz entity pages with the BookBrainz rel
+function validateBookBrainz(url) {
+  return /bookbrainz\.org\/[^\/]*\/[a-f0-9]{8}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{12}/.test(url);
```

Comment:
I think it'd be worthwhile to make the validation more precise so that `LINK_TYPES.bookbrainz.artist` can only match `/creator/` sub-paths, and etc. for the other entity types. Assuming that's all that's allowed for them.